src/main/app/controller/PostController.ts

```diff
@@ -63,6 +70,26 @@ export class PostController<T extends AnyObject> {
       res.redirect(nextUrl);
     });
   }
+
+  private async saveAndUpdateSession(
+    req: AppRequest<T>,
+    formData: Partial<Case>,
+    eventName: string
+  ): Promise<CaseWithId | undefined> {
+    try {
+      const caseData = await req.locals.api.triggerEvent(req.session.userCase.id, formData, eventName);
+
+      for (const [key, value] of Object.entries(formData)) {
+        if (!isEqual(caseData[key], value ?? '')) {
```

Comment:
why empty string?

Reply:
This is required for an unchecked checkbox as it will be `undefined` in the form data.

Reply:
I don't think this should be here, can't it go in the fromApiFormat function?

Reply:
👍 Good idea, I've moved this up into the `triggerEvent` in a6e5eca.